src/mlpack/methods/ann/dists/normal_distribution.hpp

```diff
@@ -0,0 +1,155 @@
+   * Create a Normal distribution with the given mean and sigma.
+   *
+   * @param mean The mean of the normal distribution.
+   * @param sigma The standard deviation of the normal distribution.
+   */
+  NormalDistribution(const DataType& mean, const DataType& sigma);
+
+  /**
+   * Return the probabilities of the given matrix of observations.
+   *
+   * @param observation The observation matrix.
+   */
+  DataType Probability(const DataType& observation) const
+  {
+    return arma::exp(LogProbability(observation));
+  }
+
+  /**
+   * Return the log probabilities of the given matrix of observations.
+   *
+   * @param observation The observation matrix.
+   */
+  DataType LogProbability(const DataType& observation) const;
+
+  /**
+   * Stores the gradient of the probabilities of the observations
+   * with respect to mean and standard deviation.
+   *
+   * @param observation The observation matrix.
+   * @param dmu The gradient with respect to mean.
+   * @param dsigma The gradient with respect to standard deviation.
+   */
+  void ProbBackward(const DataType& observation,
+                    DataType& dmu,
+                    DataType& dsigma) const;
+
+  /**
+   * Calculates the normal probability density function for each
+   * data point (column) in the given matrix.
+   *
+   * @param x List of observations.
+   * @param probabilities Output probabilities for each input observation.
+   */
+  void Probability(const DataType& x, DataType& probabilities) const
+  {
+    probabilities = Probability(x);
+  }
+
+  /**
+    * Calculates the log of normal probability density function for each
+    * data point (column) in the given matrix.
+    *
+    * @param x List of observations.
+    * @param log probabilities Output probabilities for each input observation.
+    */
+  void LogProbability(const DataType& x, DataType& probabilities) const
+  {
+    probabilities = LogProbability(x);
+  }
+
+  /**
+   * Return a randomly generated observation according to the probability
+   * distribution defined by this object.
+   *
+   * @return Random observation from this Normal distribution.
+   */
+  DataType Sample() const;
+
+  /**
+   * Return the mean.
+   */
+  const DataType& Mean() const { return mean; }
+
+  /**
+   * Return a modifiable copy of the mean.
+   */
+  DataType& Mean() { return mean; }
+
+  /**
+   * Return the standard deviation.
+   */
+  const DataType& StandardDeviation() const { return sigma; }
+
+  /**
+   * Return a modifiable copy of the standard deviation.
+   */
+  DataType& StandardDeviation() { return sigma; }
+
+  //! Return the dimensionality of this distribution.
+  size_t Dimensionality() const { return mean.n_elem; }
+
+  /**
+   * Serialize the distribution.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */)
+  {
+    // We just need to serialize each of the members.
+    ar & BOOST_SERIALIZATION_NVP(mean);
+    ar & BOOST_SERIALIZATION_NVP(sigma);
+  }
```

Comment:
I am sorry, it's a picky comment, Do you mind shifting this to the implementation file. Most of the code base follows that. Thanks a lot!